src/swaps/bitcoin.rs

```diff
@@ -407,18 +407,40 @@ impl BtcSwapScript {
     ) -> Result<Vec<(OutPoint, TxOut)>, Error> {
         let electrum_client = network_config.build_client()?;
         let spk = self.to_address(network_config.network())?.script_pubkey();
-        let utxos = electrum_client.script_list_unspent(spk.as_script())?;
-        let utxo_pairs = utxos
+        let history: Vec<_> = electrum_client
+            .script_get_history(spk.as_script())?
+            .into_iter()
+            .filter(|h| h.height > 0)
+            .collect();
+        let txs = electrum_client
+            .batch_transaction_get(&history.iter().map(|h| h.tx_hash).collect::<Vec<_>>())?;
```

Comment:
Nitpick: you might want to `filter` and `map` in the same statement to avoid having to collect twice